### PR TITLE
Add drop type if exists

### LINKED ISSUE #2: Add drop_type_if_exists/2
## Motivation

We need to support the if exists options of the [`DROP TYPE` statement](https://www.postgresql.org/docs/9.1/sql-droptype.html) in order to have a more complete DSL.

## Proposed solution

Introduce `drop_type_if_exists/2` function.

Below some example of the function and the generated SQL.

```elixir
drop_type_if_exists(:status)
# DROP TYPE IF EXISTS "public.status";
drop_type_if_exists(:status, schema: "custom_schema")
# DROP TYPE IF EXISTS "custom_schema.status";
```

### PR DESCRIPTION
**Pull request by antedeguemon**
Closes #2

### CODE REVIEW

File: test/ecto_enum_migration_test.exs [hunk 1]
```diff
@@ -130,6 +139,17 @@ defmodule EctoEnumMigrationTest do
         :ok = down(num, DropTypeWithCustomSchemaMigration)
       end
     end
+
+    test "supports drop if exists option" do
```
Comment (thiamsantos) [inline test/ecto_enum_migration_test.exs:143]: Could we have a another test, for the case when the type exists and it gets deleted?

Comment (antedeguemon) [reply test/ecto_enum_migration_test.exs:143]: Done @thiamsantos, thanks!

File: lib/ecto_enum_migration.ex [hunk 2]
```diff
@@ -75,7 +75,9 @@ defmodule EctoEnumMigration do
   @spec drop_type(name :: atom(), opts :: Keyword.t()) :: :ok | no_return()
   def drop_type(name, opts \\ []) when is_atom(name) and is_list(opts) do
     type_name = type_name(name, opts)
-    drop_sql = "DROP TYPE #{type_name};"
+    if_exists_sql = if_exists_sql(opts)
+
+    drop_sql = "DROP TYPE #{if_exists_sql} #{type_name};"
```
[the remark this answers is not in the document]
Comment (antedeguemon) [reply lib/ecto_enum_migration.ex:80]: Yes, we are. :grimacing: - I will fix that asap.

[the remark this answers is not in the document]
Comment (antedeguemon) [reply lib/ecto_enum_migration.ex]: @thiamsantos updated - decided to extract the joining sql terms logic into a new function, `execute_query/1`.

**Pull request merged**